cmake/modules/AddSwiftHostLibrary.cmake

```diff
@@ -6,22 +6,44 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
+# cmake generation for Swift adds an order only dependency, which matches how C-family languages
+# works. In that case, however, ninja (and presumably other generators) will rebuild on header
+# changes. That's not the case for Swift, and thus if A -> B, A is not being rebuilt when the
+# ABI/API of B changes.
+#
+# For now workaround this by touching a file whenever B is rebuilt and then compiling that file as
+# part of A. Ideally this file would be generated by each of the targets, but that dependency didn't
+# seem to be being tracked.
+#
+# Remove once rdar://102202478 is fixed.
+function(target_link_swift_syntax_libraries TARGET)
+  cmake_parse_arguments(ARGS "" "" "PUBLIC" ${ARGN})
+
+  target_link_libraries(${TARGET} PUBLIC ${ARGS_PUBLIC})
+  foreach(DEPENDENCY ${ARGS_PUBLIC})
+    add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+      COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+      DEPENDS ${DEPENDENCY}
+      )
+    target_sources(${TARGET} PRIVATE
+      ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+    )
+  endforeach()
+endfunction()
+
 # Add a new host library with the given name.
-function(add_swift_host_library name)
+function(add_swift_syntax_library name)
```

Comment:
Is this function only being used for SwiftSyntax, or will we want to pull in other things this way too?

Reply:
Oh, this is the swift-syntax repo, don't mind me!

Reply:
It's only for swift-syntax, though theoretically we could replace the swift functions with functions defined here instead. 🤷